Update to es6

CarlosZ96 opened the pull request:
HTML, CSS and JavaScript file created, Module book created, class Abooks is export and import inside the module index

CarlosZ96 committed:
Update README.md

README FILE TEST

CarlosZ96 committed:
Update README.md

Minor readme changes

CarlosZ96 committed:
Update README.md

All tex changed.

CarlosZ96 committed:
Update package.json

Update package.json

CarlosZ96 committed:
Creating files

HTML, CSS and JavaScript file created

CarlosZ96 committed:
Merge branch 'Update-to-ES6' of https://github.com/CarlosZ96/Awesome-Books-ES6 into Update-to-ES6

CarlosZ96 committed:
Create settings.json

Create settings.json

CarlosZ96 committed:
Delete MIT.md

Delete MIT

CarlosZ96 committed:
Create LICENSE

Create LICENSE

CarlosZ96 committed:
Update index.js

Import luxon

CarlosZ96 committed:
Merge branch 'Update-to-ES6' of https://github.com/CarlosZ96/Awesome-Books-ES6 into Update-to-ES6

CarlosZ96 committed:
Update

Update for HTML and CSS add skeleton and styles

CarlosZ96 committed:
Update

Module book created, class Abooks is export and import inside the module index

CarlosZ96 committed:
Fix

removing unnecessary files.

CarlosZ96 committed:
Update README.md

Update the Readme File

CarlosZ96 committed:
Create linters.yml

Create linters.yml

CarlosZ96 committed:
Update linters.yml

Update linters.yml

CarlosZ96 committed:
Update book.js

Fix linterns

CarlosZ96 committed:
Merge branch 'Update-to-ES6' of https://github.com/CarlosZ96/Awesome-Books-ES6 into Update-to-ES6

CarlosZ96 committed:
Update index.js

Import position changed.

CarlosZ96 committed:
Fix

Fix Import and Export.

CarlosZ96 committed:
Update index.js

Time changed.

CarlosZ96 committed:
Update index.js

Add Time from Los Angeles

CarlosZ96 committed:
Update index.js

Minor Fix.

CarlosZ96 committed:
Update index.js

Minor Fix.

CarlosZ96 committed:
Update index.js

Minor Fix.

julie-ify commented on index.html:
```diff
@@ -0,0 +1,71 @@
+      </nav>
+      <main class="list">
+        <div class="date">
+          <p class="today-date">May 5th 2023, 11:57</p>
```
- [ ] Your date is static. Kindly note that you should use the luxon package from the javascript to dynamically render the real date, instead of having it this way.

- [ ] **[OPTIONAL]** I suggest you stretch the copyright section to the button of the pages, instead of having it midway through the page.

![image](https://github.com/CarlosZ96/Awesome-Books-ES6/assets/81468675/9deb2a1f-c4aa-4439-a155-eaa966a389e6)

julie-ify reviewed (changes requested):
## Hi @CarlosZ96,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

CarlosZ96 committed:
Luxon

Creating everything to use Luxon.

CarlosZ96 committed:
fix

Removing unnecessary files.

CarlosZ96 committed:
Fix

Add Luxon Library

CarlosZ96 committed:
Create .gitignore

Adding the node_module to Gitignore

CarlosZ96 committed:
Update .gitignore

Add node_modules

CarlosZ96 committed:
Update .gitignore

fix

CarlosZ96 committed:
Update .gitignore

Fix

CarlosZ96 committed:
Luxon

Try to put node_modules in gitignore file

CarlosZ96 committed:
Luxon

The luxon.js file is changed, the section for the clock with the date is added.

CarlosZ96 committed:
Update .gitignore

Update gitignore

CarlosZ96 committed:
Update .gitignore

Update Gitignore

CarlosZ96 committed:
Delete luxon.js

Delete luxon

CarlosZ96 committed:
Luxon

Fix

CarlosZ96 committed:
Update .gitignore

Fix

CarlosZ96 committed:
Update .gitignore

fix

CarlosZ96 committed:
Update luxon.js

fix

CarlosZ96 committed:
Delete luxon.js

Delte Luxon

CarlosZ96 committed:
Update .gitignore

Try to fix gitignore.

CarlosZ96 committed:
Fix

Fix linters errors

CarlosZ96 committed:
Update .eslintrc.json

solve module issues

CarlosZ96 committed:
Update index.css

Copyright resizing

CarlosZ96 committed:
Fix linterns error

Fix linters error

CarlosZ96 committed:
Fix

Trying to solve the problem with luxon.js

CarlosZ96 committed:
Fix linters

Trying to solve the problem with luxon.js

CarlosZ96 committed:
Fix linters

Trying to solve this problem with the luxon.js file

CarlosZ96 committed:
Update .eslintrc.json

Trying to solve this problem with the luxon.js file

CarlosZ96 committed:
Delete linters.yml

Trying to solve a problem with luxon.js

CarlosZ96 committed:
Create linters.yml

Tratando de resolver un problema con la luxon.js

CarlosZ96 committed:
Update index.js

Trying to solve a problem with luxon.js

CarlosZ96 committed:
Merge branch 'Update-to-ES6' of https://github.com/CarlosZ96/Awesome-Books-ES6 into Update-to-ES6

CarlosZ96 committed:
Update index.js

Tratando de resolver un problema con la luxon.js

Munsa1 commented on modules/Time.js:
```diff
@@ -0,0 +1,6 @@
+/* eslint-disable import/prefer-default-export */
+export const gettime = (() => {
+  const time = document.querySelector('.date');
+  const ctimep = document.createElement('p');
+  time.append(ctimep);
```
- [ ] As requested by the previous reviewer, kindly use [luxon library](https://github.com/moment/luxon/blob/master/docs/install.md) to display the time below the navigation bar as shown in the given wireframe

Wireframe 
![image](https://github.com/CarlosZ96/Awesome-Books-ES6/assets/81932478/c6043a7b-bec6-4951-8380-8ede7df2e561)

Your design
![image](https://github.com/CarlosZ96/Awesome-Books-ES6/assets/81932478/a74fa827-6bb8-4cec-a5a5-eced652f1d68)

Munsa1 commented on modules/book.js:
```diff
@@ -0,0 +1,36 @@
+      }
+    };
+
+    this.displayBooks = () => {
+      if (localStorage.getItem('books')) {
+        books = JSON.parse(localStorage.getItem('books'));
+      }
+      bookList.className = 'bookList';
+      bookList.innerHTML = '';
+      books.forEach((book, index) => {
+        const li = document.createElement('li');
+        li.className = 'lineBook';
+        li.innerHTML = ` " ${book.title} " by ${book.author} <button class="removeBtn" data-index="${index}">Remove</button>`;
+        bookList.appendChild(li);
+      });
+    };
+
+    this.submitbtn = () => {
+      const title = document.getElementById('bookTitle').value;
+      const author = document.getElementById('authorName').value;
```
- [ ] Something has broken the code, the user is no longer able to add and delete books. Kindly fix this

Munsa1 reviewed (changes requested):
Hi @CarlosZ96, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
Used ES6  JS :heavy_check_mark: 
Used Modular Programming :heavy_check_mark: 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Munsa1 reviewed (changes requested):
Hi @CarlosZ96, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
Used ES6  JS :heavy_check_mark: 
Used Modular Programming :heavy_check_mark: 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

CarlosZ96 committed:
Deploy Update

Add URL

CarlosZ96 committed:
Update index.html

Fix HTML

CarlosZ96 committed:
Fix

Fix Deploy error

CarlosZ96 committed:
Deploy

Add modules file to deploy

CarlosZ96 committed:
Update package.json

Fix

CarlosZ96 committed:
Update .gitignore

Fix

CarlosZ96 committed:
Create luxon.js

Fix Luxon error

CarlosZ96 committed:
Update

Add clock for all sections and minor style changes

CarlosZ96 committed:
Update index.css

Minor fix

ridwanediallo commented on index.html:
```diff
@@ -0,0 +1,62 @@
+      <ul class="book-list" id="bookList"></ul>
+    </div>
+    </div>
+    <div class="copyright-div">
+      <p class="copyright">Copyright .. </p>
+    </div>
```
- [ OPTIONAL] For semantic reasons, I recommend putting this below and wrapping it with the `footer` tag rather than a `div`. And to fix it at the bottom of the page, I recommend updating the style as shown below. :point_down: 

```
.copyright  {
    border: 3px solid black;
    padding: 10px;
    position: fixed;
    bottom: 0px;
    left: 0px;
    right: 0px;
    margin: 5px 20px;
}

```
#### Your result will look like this :point_down: 
![Screenshot from 2023-05-11 16-59-09](https://github.com/CarlosZ96/Awesome-Books-ES6/assets/89228743/e17d0008-0640-4e1d-b03f-b609541b1a7d)



#### Currently this :point_down:  is what we have.

![Screenshot from 2023-05-11 16-54-32](https://github.com/CarlosZ96/Awesome-Books-ES6/assets/89228743/9942ed42-1b97-4922-88bb-53dec670ae6e)

ridwanediallo reviewed (approved):
# STATUS: APPROVED 🟢
## Hi <STUDENT_MENTION>, 
![Congratulations!GIF](https://user-images.githubusercontent.com/55918904/191301390-43d62dfa-bfdc-49c5-943b-4d72f4ea5bb8.gif)

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Well done implementing the requested changes. ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

IngDavidTM reviewed (commented):
Hi @CarlosZ96 🙋‍♂️, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request follows the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 
<img src="https://media3.giphy.com/media/xT0xezQGU5xCDJuCPe/giphy.gif"/>

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
